cli/azd/pkg/pipeline/github_provider.go

```diff
@@ -627,18 +619,100 @@ func (p *GitHubCiProvider) configurePipeline(
 	provisioningProvider provisioning.Options,
 	additionalSecrets map[string]string,
 	additionalVariables map[string]string,
+	options *ConfigOptions,
 ) (CiPipeline, error) {
-
 	repoSlug := repoDetails.owner + "/" + repoDetails.repoName
+
+	// Variables and Secrets for a gh-actions are independent from the gh-action. They are set on the repository level.
+	// We need to clean up the previous values before setting the new ones.
+	// By doing this, we are handling:
+	// - When a secret is moved to be a variable (or vice versa). Don't leak the previous value on the pipeline.
+	// - When there was a previous additional variable/secret set and then it was updated to empty string or unset from .env.
+	msg := ""
+	var procErr error
+	ciSecrets, ciVariables := []string{}, []string{}
+	if len(options.Variables) > 0 || len(options.Secrets) > 0 {
+		msg = "Setting up project's secrets and variables to be used in the pipeline"
+		ciSecretsInstance, err := p.ghCli.ListSecrets(ctx, repoSlug)
+		if err != nil {
+			return nil, fmt.Errorf("unable to get list of repository secrets: %w", err)
+		}
+		ciVariablesInstance, err := p.ghCli.ListVariables(ctx, repoSlug)
+		if err != nil {
+			return nil, fmt.Errorf("unable to get list of repository variables: %w", err)
+		}
+		ciSecrets = ciSecretsInstance
+		ciVariables = ciVariablesInstance
+		p.console.ShowSpinner(ctx, msg, input.Step)
```

Comment:
This may be a minor nit. Suppose that no `variables` or `secrets` were defined in `azure.yaml`.

However, `AZURE_RESOURCE_GROUP` and an environment config is present. In these cases, there are indeed secrets and variables to set. Yet, we won't display the spinner.

Reply:
Correct.  
I tried to move all the default-env-vars that ci-providers are currently setting, like rg, location, azd-environment-name, etc. But there are some complicated changes that makes the PR to grow in another direction.
So, I decided to do that as a separate change. At any case, the other variables you mention are individually listed in the output.